### DISCUSSION TITLE
Next build window not definied

### DISCUSSION THREAD

**Discussion by avmentzer** [vercel/next.js#50508] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.16.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.4.5-canary.0
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Dynamic imports (next/dynamic)

### Link to the code that reproduces this issue or a replay of the bug

none

### To Reproduce

`
import Home from '@/components/body/home/page';
import { getAuth } from '@/components/helpers/auth';
import dynamic from 'next/dynamic';
import { Metadata } from 'next';

const Map = dynamic(() => import('./map/page'), { ssr: false })

export const metadata: Metadata = {
  title: '...',
};

export default async function Body() {

  if (await getAuth()) {
    return (
      <div>
        <Map />
      </div>
    )
  } else {
    return (
      <Home />
    )
  }
}
`

### Describe the Bug

I have a react-leaflet page which I import in the code seen above. That works fine and the page renders with no errors.  when I access the root page "/".

But when I try to build the project "/map/page" fails at generating static sites because 

"
- error node_modules\leaflet\dist\leaflet-src.js (230:0) @ eval
- error ReferenceError: window is not defined
- "

This error also occurs when I access the page directly or when I create an component outside of /app.
I need this project to be done asap and I'm wondering if there is some workaround?

### Expected Behavior

No problem at build time.

**Comment by GabenGar** [comment]
Render the entire component with leaflet conditionally by checking for presence of `window`:
```
if (typeof window === "undefined) {
  return <span>Loading...</span>
}

return ...
```
Assuming `react-leaflet` doesn't do browser-related side-effects right at the top level of its modules.
